### PR TITLE
Change representation of boxes

### PR DESCRIPTION
**Pull request by degustaf**
This is a first step to supporting irregular regions. Boxes are now represented as an idxArray.

### CODE REVIEW

File: packages/board-utils/src/board-repr.ts [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+    const regions: ArrayObj<IdxBitset<Geometry.CELL>> = {};
+    for (let i = 0; i < size; i++) {
+        regions[i] = {};
+    }
+
+    for (let y = 0; y < size; y++) {
+        for (let x = 0; x < size; x++) {
+            // Inlined to prevent circular dependency.
+            // const grid: schema.Grid = { width: size, height: size };
+            // const cellIdx = cellCoord2CellIdx([ x, y ], grid);
```
Comment (wooferzfg) [inline packages/board-utils/src/board-repr.ts:20]: Can you remove this? (I think we generally prefer to not have commented out code)

**Comment by degustaf** [comment]
Fixed a bug for non-square regions, and removed the commented code.

File: packages/board/src/svelte/BoxRender.svelte [hunk 2]
```diff
@@ -1,18 +1,21 @@
 <script lang="ts">
-    import { BOX_THICKNESS, BOX_THICKNESS_HALF } from "@sudoku-studio/board-utils";
+    import { arrayObj2array, getBorderPath, idxMapToKeysArray, BOX_THICKNESS, BOX_THICKNESS_HALF } from "@sudoku-studio/board-utils";
     import type { StateRef } from "@sudoku-studio/state-manager";
 
     export let id: string;
-    export let ref: StateRef;
+    export let ref: StateRef;    // schema.BoxElement
```
Comment (wooferzfg) [inline packages/board/src/svelte/BoxRender.svelte:6]: Would it make sense to rename this to `RegionRender`? And maybe also renaming the constants being imported from `board-utils` to `REGION_BORDER_THICKNESS` and `REGION_BORDER_THICKNESS_HALF` as well?

Comment (degustaf) [reply packages/board/src/svelte/BoxRender.svelte:6]: Sure. should I rename `BoxInfo` and `BoxElement` as well?

Comment (wooferzfg) [reply packages/board/src/svelte/BoxRender.svelte:6]: Yeah, I think that'd be great

Comment (degustaf) [reply packages/board/src/svelte/BoxRender.svelte:6]: My plan was to rename `BoxElement` to `RegionElement`, but that already exists for max, min, even, and odd. Any thoughts on a better name?

Comment (wooferzfg) [reply packages/board/src/svelte/BoxRender.svelte:6]: Maybe `GridRegionElement` or `SudokuRegionElement`?

**Comment by degustaf** [comment]
Renaming complete

**Pull request merged**